File: C_Plus_Plus/1001-2000/p1207.cpp
#include <vector>
#include <unordered_map>
#include <unordered_set>

using namespace std;

class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> mp;
        unordered_set<int> hash;
        for(int num : arr){
            mp[num]++;
        }
        for(const auto& [key, value] : mp){
            if(hash.contains(value)){
                return false;
            }
            hash.insert(value);
        }
        return true;
    }
};

File: C_Sharp/1001-2000/p1207.cs
namespace p1207 {
    public class Solution {
        public bool UniqueOccurrences(int[] arr) {
            Dictionary<int,int> mp = new Dictionary<int, int> ();
            HashSet<int>hash = new HashSet<int>();
            foreach (int num in arr)
            {
                mp.TryGetValue(num, out int curr);
                mp[num] = curr + 1;
            }
            return mp.Select(x => x.Value).Distinct().ToArray().Length == mp.Count();
        }
    }    
}